lib/components/base-components/PrimitiveComponent.ts

```diff
@@ -49,6 +49,18 @@ export abstract class PrimitiveComponent<
 
   externallyAddedAliases: string[]
 
+  getPropNetReferences(): { [key: string]: string } {
+    return {}
+  }
+
+  doInitialCreateNetsFromProps() {
+    const netReferences = this.getPropNetReferences()
+    for (const [propName, netName] of Object.entries(netReferences)) {
+      const net = new Net({ name: netName })
+      this.add(net)
```

Comment:
aider: don't add the net to this element, use `this.getOpaqueGroup().add(net)`, also check that the net doesn't already exist, maybe by introducing a method `checkHasNet(netName)`